pkg/prometheus_ingester/prometheus_ingester.go

```diff
@@ -75,9 +76,48 @@ type queryOptions struct {
 	ResultAsQuantity *bool
 }
 
+type PrometheusIngesterHttpHeader struct {
+	Name            string
+	ValueFromString string `yaml:"valueFromString"`
+	ValueFromEnv    string `yaml:"valueFromEnv"`
+	Value           string `yaml:"-"`
+}
+
+func (h *PrometheusIngesterHttpHeader) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	type PrometheusIngesterHttpHeaderCustom PrometheusIngesterHttpHeader
+
+	out := PrometheusIngesterHttpHeaderCustom{}
+
+	if err := unmarshal(&out); err != nil {
+		return err
+	}
+
+	*h = PrometheusIngesterHttpHeader(out)
+
+	if (h.ValueFromEnv == "") == (h.ValueFromString == "") {
+		return fmt.Errorf("exactly one of 'ValueFromString' or 'ValueFromEnv' must be set %+v", h)
+	}
+
+	if h.ValueFromString != "" {
+		h.Value = h.ValueFromString
+	}
+
+	if h.ValueFromEnv != "" {
+		value, ok := os.LookupEnv(h.ValueFromEnv)
+		if !ok {
+			return fmt.Errorf("value of environment variable '%s' is empty", h.ValueFromEnv)
```

Comment:
The error message is wrong.
```suggestion
			return fmt.Errorf("environment variable '%s' is not set", h.ValueFromEnv)
```

Reply:
fixed d48902d4ad77ea48e830ce5ce24c2a6aa5fce662